evaluator/package_cache.go

```diff
@@ -89,82 +115,182 @@ func (c *PackageCache) Load() {
 			DescriptionHash: columns.DescriptionHash,
 			SummaryHash:     columns.SummaryHash,
 		}
-		c.AddWithoutLock(&pkg)
+		c.Add(&pkg)
 		count++
 	}
 	progressTicker.Stop()
 
 	runtime.ReadMemStats(&mEnd)
-	utils.Log("rows", len(c.byID), "allocated-size", utils.SizeStr(mEnd.TotalAlloc-mStart.TotalAlloc),
+	utils.Log("rows", c.byID.Len(), "allocated-size", utils.SizeStr(mEnd.TotalAlloc-mStart.TotalAlloc),
 		"duration", utils.SinceStr(tStart, time.Millisecond)).Info("PackageCache.Load")
 }
 
 func (c *PackageCache) GetByID(id int) (*PackageCacheMetadata, bool) {
-	c.mtx.RLock()
-	defer c.mtx.RUnlock()
-	metadata, ok := c.byID[id]
-	utils.Log("id", id, "ok", ok).Trace("PackageCache.GetByID")
-	if ok {
+	if c.enabled {
```

Comment:
Default go style is to use early return in case of nested ifs.

Reply:
I'm not sure what you mean. How would you propose to change it?